parsec/mca/device/cuda/device_cuda_module.c

```diff
@@ -1765,8 +1765,8 @@ parsec_gpu_callback_complete_push(parsec_device_gpu_module_t   *gpu_device,
         assert( flow );
         assert( flow->flow_index == i );
         if(PARSEC_FLOW_ACCESS_NONE == (PARSEC_FLOW_ACCESS_MASK & flow->flow_flags)) continue;
-        if(PARSEC_DATA_STATUS_UNDER_TRANSFER == task->data[i].data_out->data_transfer_status ) {
-            assert(task->data[i].data_out->push_task == task );  /* only the task who did this PUSH can modify the status */
+        if( PARSEC_DATA_STATUS_UNDER_TRANSFER == task->data[i].data_out->data_transfer_status ) {
+            /* only the task who did the PUSH can modify the status */
```

Comment:
Is this comment still relevant? The associated assert has been removed and the following doesn't seem to be related.

Reply:
the comment is still valid but we do not have a direct way to know who was the task that initiated the transfer. However, as the transfers in each direction are sequentialized for each GPU, the first task that completes its transfer and has any of the data marked as "under transfer" must be the task that initiated it and thus the only one allowed to change it.